### DISCUSSION TITLE
Can’t generate Installation Access Token

### DISCUSSION THREAD

**Discussion by paul-carron** [community/community#21904] [New to GitHub]
<p>I followed <a href="https://developer.github.com/apps/building-github-apps/authentication-options-for-github-apps/" rel="nofollow noopener">this</a> documentation to generate an installation access token but I getht this response:</p>
<p>{<br>
“message”: “Not Found”,<br>
“documentation_url”: “<a href="https://developer.github.com/v3/apps/#create-a-new-installation-token" rel="nofollow noopener">https://developer.github.com/v3/apps/#create-a-new-installation-token</a>”<br>
}</p>
<p>I have a feeling the issue might be with my instalaltion_id. Can anybody confirm where I should get this?</p>

**Comment by ThirdEyeRose** [comment]
<p>Hi <a class="mention" href="/paul-carron">@paul-carron</a>,</p>
<p>This post was moved to a different board that fits your topic of discussion a bit better. This means you’ll get better engagement on your post, and it keeps our Community organized so users can more easily find information.</p>
<p>As you’ll notice, your Topic is now in the How to use Git and GitHub board. No action is needed on your part; you can continue the conversation as normal here.</p>
<p>Let me know if you have any other questions or if I can help with anything else.</p>
<p>Cheers!</p>

**Comment by paul-carron** [comment]
<p>curl -i -X POST \<br>
&gt; -H “Authorization: Bearer token” \<br>
&gt; -H “Accept: application/vnd.github.machine-man-preview+json” \<br>
&gt; <a href="https://api.github.com/installations/13631156/access_tokens" rel="nofollow noopener">https://api.github.com/installations/13631156/access_tokens</a><br>
HTTP/1.1 404 Not Found<br>
Server: <a href="http://GitHub.com" rel="nofollow noopener">GitHub.com</a><br>
Date: Mon, 12 Mar 2018 13:08:56 GMT<br>
Content-Type: application/json; charset=utf-8<br>
Content-Length: 125<br>
Status: 404 Not Found<br>
X-GitHub-Media-Type: github.machine-man-preview; format=json<br>
Access-Control-Expose-Headers: ETag, Link, Retry-After, X-GitHub-OTP, X-RateLimit-Limit, X-RateLimit-Remaining, X-RateLimit-Reset, X-OAuth-Scopes, X-Accepted-OAuth-Scopes, X-Poll-Interval<br>
Access-Control-Allow-Origin: *<br>
Content-Security-Policy: default-src ‘none’<br>
Strict-Transport-Security: max-age=31536000; includeSubdomains; preload<br>
X-Content-Type-Options: nosniff<br>
X-Frame-Options: deny<br>
X-XSS-Protection: 1; mode=block<br>
X-Runtime-rack: 0.017574<br>
X-GitHub-Request-Id: FCDF:33E6:845AB63:F3253A1:5AA67BE8</p>
<p>{<br>
  “message”: “Not Found”,<br>
  “documentation_url”: “<a href="https://developer.github.com/v3/apps/#create-a-new-installation-token" rel="nofollow noopener">https://developer.github.com/v3/apps/#create-a-new-installation-token</a>”<br>
}</p>
<p>There you go.</p>
<p>I suspect I may not have the correct installation_id in my URL. I used owner.id which I got from here:</p>
<p>curl -i -H “Authorization: Bearer token” -H “Accept: application/vnd.github.machine-man-preview+json” <a href="https://api.github.com/app" rel="nofollow noopener">https://api.github.com/app</a><br>
HTTP/1.1 200 OK<br>
Server: <a href="http://GitHub.com" rel="nofollow noopener">GitHub.com</a><br>
Date: Mon, 12 Mar 2018 12:47:44 GMT<br>
Content-Type: application/json; charset=utf-8<br>
Content-Length: 1389<br>
Status: 200 OK<br>
Cache-Control: public, max-age=60, s-maxage=60<br>
Vary: Accept<br>
ETag: “3e5af20afa747c7ee416bc063a413a04”<br>
X-GitHub-Media-Type: github.machine-man-preview; format=json<br>
Access-Control-Expose-Headers: ETag, Link, Retry-After, X-GitHub-OTP, X-RateLimit-Limit, X-RateLimit-Remaining, X-RateLimit-Reset, X-OAuth-Scopes, X-Accepted-OAuth-Scopes, X-Poll-Interval<br>
Access-Control-Allow-Origin: *<br>
Content-Security-Policy: default-src ‘none’<br>
Strict-Transport-Security: max-age=31536000; includeSubdomains; preload<br>
X-Content-Type-Options: nosniff<br>
X-Frame-Options: deny<br>
X-XSS-Protection: 1; mode=block<br>
X-Runtime-rack: 0.020248<br>
X-GitHub-Request-Id: FA40:33E4:3F632C5:8358375:5AA676F0</p>
<p>{<br>
  “id”: xxxx,<br>
  “owner”: {<br>
    “login”: “MyOrg”,<br>
    “id”: xxxx,<br>
    “avatar_url”: “<a href="https://avatars3.githubusercontent.com/u/xxxx?v=4" rel="nofollow noopener">https://avatars3.githubusercontent.com/u/xxxx?v=4</a>”,<br>
    “gravatar_id”: “”,<br>
    “url”: “<a href="https://api.github.com/users/PCTestOrg" rel="nofollow noopener">https://api.github.com/users/PCTestOrg</a>”,<br>
    “html_url”: “<a href="https://github.com/PCTestOrg" rel="nofollow noopener">https://github.com/PCTestOrg</a>”,<br>
    “followers_url”: “<a href="https://api.github.com/users/PCTestOrg/followers" rel="nofollow noopener">https://api.github.com/users/PCTestOrg/followers</a>”,<br>
    “following_url”: “<a href="https://api.github.com/users/PCTestOrg/following%7B/other_user" rel="nofollow noopener">https://api.github.com/users/PCTestOrg/following{/other_user</a>}”,<br>
    “gists_url”: “<a href="https://api.github.com/users/PCTestOrg/gists%7B/gist_id" rel="nofollow noopener">https://api.github.com/users/PCTestOrg/gists{/gist_id</a>}”,<br>
    “starred_url”: “<a href="https://api.github.com/users/PCTestOrg/starred%7B/owner%7D%7B/repo" rel="nofollow noopener">https://api.github.com/users/PCTestOrg/starred{/owner}{/repo</a>}”,<br>
    “subscriptions_url”: “<a href="https://api.github.com/users/PCTestOrg/subscriptions" rel="nofollow noopener">https://api.github.com/users/PCTestOrg/subscriptions</a>”,<br>
    “organizations_url”: “<a href="https://api.github.com/users/PCTestOrg/orgs" rel="nofollow noopener">https://api.github.com/users/PCTestOrg/orgs</a>”,<br>
    “repos_url”: “<a href="https://api.github.com/users/PCTestOrg/repos" rel="nofollow noopener">https://api.github.com/users/PCTestOrg/repos</a>”,<br>
    “events_url”: “<a href="https://api.github.com/users/PCTestOrg/events%7B/privacy" rel="nofollow noopener">https://api.github.com/users/PCTestOrg/events{/privacy</a>}”,<br>
    “received_events_url”: “<a href="https://api.github.com/users/PCTestOrg/received_events" rel="nofollow noopener">https://api.github.com/users/PCTestOrg/received_events</a>”,<br>
    “type”: “Organization”,<br>
    “site_admin”: false<br>
  },<br>
  “name”: “IBM Stocator Test Runner - DEV”,<br>
  “description”: “This is a DEV app for running multiple test types against the IBM Stocator connector.”,<br>
  “external_url”: “<a href="https://github.com/PCTestOrg/stocator" rel="nofollow noopener">https://github.com/PCTestOrg/stocator</a>”,<br>
  “html_url”: “<a href="https://github.com/apps/ibm-stocator-test-runner-dev" rel="nofollow noopener">https://github.com/apps/ibm-stocator-test-runner-dev</a>”,<br>
  “created_at”: “2018-02-22T12:47:16Z”,<br>
  “updated_at”: “2018-03-12T12:05:17Z”<br>
}</p>
<p>If this is incorrect, where should the installation_id come from?</p>

**Comment by paul-carron** [comment, marked as answer]
<p>In order to list your application installations, including the installation ID used for this endpoint, use the /app/installations endpoint documented here: <a href="https://developer.github.com/v3/apps/#find-installations" rel="nofollow noopener">https://developer.github.com/v3/apps/#find-installations</a></p>
